core/generator/generator.go

```diff
@@ -59,6 +60,8 @@ func Generate(
 		log.Fatal(ctx, log.KeyError, err)
 	}
 
+	networkReady()
```

Comment:
Instead of a callback here maybe we should have a `close(c.ready)` at the bottom of protocol.Chain.Recover, plus an accessor method for that chan.

So in main, we would do

```
go func() {
	<-c.Ready()
	height := c.Height()
	...
}()
```

That would avoid the extra plumbing in generator and fetch right? WDYT?